File: packages/sdk-python/src/eidolondb/resources/context.py
from __future__ import annotations

from typing import Any

from .._client import EidolonDBClient
from .._types import ContextBuildResponse


class ContextResource:
    def __init__(self, client: EidolonDBClient):
        self._client = client

    def build(self, current_input: str, **kwargs: Any) -> ContextBuildResponse:
        payload = {"currentInput": current_input, **kwargs}
        return self._client.request("POST", "/context/build", body=payload)

File: packages/sdk-js/src/resources/context.ts
import { EidolonDBClient } from '../client.js';
import type { ContextBuildInput, ContextBuildResponse } from '../types.js';

export class ContextResource {
  constructor(private readonly client: EidolonDBClient) {}

  /** Build LLM-ready context from memories and current input. */
  build(input: ContextBuildInput): Promise<ContextBuildResponse> {
    return this.client.request<ContextBuildResponse>('POST', '/context/build', input);
  }
}
